sycl/source/detail/config.hpp

```diff
@@ -0,0 +1,103 @@
+//==---------------- config.hpp - SYCL context ------------------*- C++-*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cstdlib>
+
+namespace cl {
+namespace sycl {
+namespace detail {
+
+#ifdef DISABLE_CONFIG_FROM_ENV
+constexpr bool ConfigFromEnvEnabled = false;
+#else
+constexpr bool ConfigFromEnvEnabled = true;
+#endif // DISABLE_CONFIG_FROM_ENV
+
+#ifdef DISABLE_CONFIG_FROM_CONFIG_FILE
+constexpr bool ConfigFromFileEnabled = false;
+#else
+constexpr bool ConfigFromFileEnabled = true;
+#endif // DISABLE_CONFIG_FROM_CONFIG_FILE
+
+#ifdef DISABLE_CONFIG_FROM_COMPILE_TIME
+constexpr bool ConfigFromCompileDefEnabled = false;
+#else
+constexpr bool ConfigFromCompileDefEnabled = true;
+#endif // DISABLE_CONFIG_FROM_COMPILE_TIME
+
+// Enum of config IDs for accessing other arrays
+enum ConfigID {
+  START = 0,
+#define CONFIG(name, ...) name,
+#include "config.def"
+#undef CONFIG
+  END
+};
+
+// Consider strings starting with __ as unset
+constexpr const char *getStrOrNullptr(const char *Str) {
+  return (Str[0] == '_' && Str[1] == '_') ? nullptr : Str;
```

Comment:
What happened if a string has less than 2 characters?

Reply:
Will add check for that.